### PR TITLE
Repo selinux context

### PR DESCRIPTION
**Pull request by mattohope**
Added selinux context support to the repository creation

### CODE REVIEW

File: manifests/svnrepo.pp [hunk 1]
```diff
@@ -66,5 +67,12 @@
     File[$create_path]{
       mode => $mode,
     }
-  } 
+  }
+  # if selinux is enabled and there is a required context apply it
+  if $selinux_context and str2bool($::selinux) {
+    exec {"set_${name}_repo_selinux_context":
+      command     => "semanage fcontext -a -t ${selinux_context} \"${$create_path}(/.*)?\" && restorecon -R -v ${$create_path}",
+      require     => File[$create_path],
```
Comment (duritong) [inline manifests/svnrepo.pp]: can we make that idempotent or is it the idea that this exec will be executed each time.

Comment (mattohope) [inline manifests/svnrepo.pp]: no we can make it idempotent, my thought would be to add something like this:
unless  => "semanage fcontext -l | grep ${$create_path}(/._)?._${selinux_context}:"

Comment (duritong) [inline manifests/svnrepo.pp]: Sounds good to me

Comment (duritong) [inline manifests/svnrepo.pp]: Actually, we could also make it just refreshonly, because we likely need to do that only after we did the checkout.

Comment (mattohope) [inline manifests/svnrepo.pp]: ok i'll change the exec and add the refreshonly option and subscribe it to the repo folder

**Comment by mattohope** [comment]
selinux exec is now refreshonly

**Comment by duritong** [comment]
thanks!

**Pull request merged**